src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -96,6 +97,16 @@ public void writeAnnotation_withDescription() {
     assertEquals(writerVisitor.write(), "@SuppressWarnings(\"all\")\n");
   }
 
+  @Test 
+  public void writeNewObjectValue_withArgs(){
+      Reference reference = Reference.withClazz(Integer.class);
+      TypeNode type = TypeNode.withReference(reference);
+      ValueExpr valueExpr = ValueExpr.builder().setValue(PrimitiveValue.builder().setType(TypeNode.INT).setValue("123").build()).build();
+      NewObjectValue newObjectValue = NewObjectValue.builder().setType(type).setArguments(Arrays.asList(valueExpr)).build();
+      newObjectValue.accept(writerVisitor);
+      assertEquals(writerVisitor.write(), "new Integer(123)");
+  }
+
```

Comment:
can we have assignment test like `List<Book> books = new ArrayList<Book>(); ` to build confidence of this feature.

Reply:
yes sure, once the design is approved, I will add more tests like you mentioned. Thank you!

Reply:
This kind of assignment is included in the unit test, thanks!